src/Build/BackEnd/Components/ProjectCache/ProjectCacheService.cs

```diff
@@ -582,6 +582,12 @@ public async Task ShutDown()
                 _loggingService,
                 BuildEventContext.Invalid,
                 BuildEventFileInfo.Empty);
+            bool shouldInitiateShutdownState = _serviceState != ProjectCacheServiceState.ShutdownStarted && _serviceState != ProjectCacheServiceState.ShutdownFinished;
```

Comment:
Move this up a few lines? Not sure why we should initialize a LoggingServiceToPluginLoggerAdapter if we might just return anyway.